Automattic/jetpack
# do_full_sync should early return if JETPACK_SYNC_READ_ONLY is true

In #17119 we introduced the constant JETPACK_SYNC_READ_ONLY to allow for disabling sync sends during various requests. During the rollout we added it to the incremental sync but not for full_sync actions. This PR updates the do_full_sync handler to ensure it doesn't send data when in READ ONLY mode.

#### Changes proposed in this Pull Request:
* Disable full_sync sends if in READ_ONLY mode.

#### Does this pull request change what data or activity we track or use?
* NO

#### Testing instructions:

* define( 'JETPACK_SYNC_READ_ONLY', true ); on your Jetpack Site.
* Using the Jetpack Debugger, CLI, etc trigger a Full Sync.
* Click around the site (perform various requests)
* Check the status of the Full Sync it should not have sent data.
* Remove the JETPACK_SYNC_READ_ONLY constant..
* Click around the site
* Full Sync will be sending data to WP.com

#### Proposed changelog entry for your changes:
* Jetpack Sync :: Disable Full-Syncs when in READ ONLY mode.
